### DISCUSSION TITLE
Starting Rustdesk automatically on Windows

### DISCUSSION THREAD

**Discussion by henriquelucas** [rustdesk/rustdesk#3930] [Show and tell]
I hope this is useful for other people.

Recently install Rustdesk Server on a Windows and maybe there are other ways to improve this. But to run the server automatically create a simple .bat for this.

```
echo Open RustDesk Automatically
cd C:\Program Files\RustDeskServer\bin\
start hbbr.exe
start hbbs.exe
```
And I added the bat in the Windows startup folder, if you don't know where this folder is, just add the
just open the run keyboard `Windows + R`
and modify shell:startup then just put the bat inside the folder, every time windows restarts, when turned on it will start your server Rustdesk Server

**Comment by Scampi-ml** [comment]
If you have the service installed, there is no need for this

**Reply by Karmalakas** [reply]
How do I install the service? I managed to find instructions somewhere months ago, but now setting up a PC from scratch and can't figure out how to do that 🙁

**Reply by Karmalakas** [reply]
Hmm.. Found [the article](https://rustdesk.com/docs/en/self-host/rustdesk-server-oss/windows/), did everything seemingly as supposed, both hbbs and hbbr services are running even after restart, but I still need to open RustDesk main app to be able to connect 😕

**Reply by steveace60** [reply]
Here is a Windows guide I followed that worked for me.

https://www.reddit.com/r/rustdesk/comments/1aenxrj/is_there_a_comprehensive_noobfriendly_tutorial/

Open the required TCP and UDP ports in windows firewall and set up port forwarding in your router for public access.  I have a dynamic ip address so I use a [No-IP](https://www.noip.com/) Dynamic Update Client (DUC) to sort that out.

**Comment by xlionjuan** [comment]
Install it as service, or start it though Windows Task Scheduler.

**Reply by Karmalakas** [reply]
How?

[Here's](https://github.com/rustdesk/rustdesk/discussions/3930#discussioncomment-8236392) what I did

**Reply by cdongtao** [reply]
how about the Mac mini ? I can not run up automatically while it start up.

**Reply by xlionjuan** [reply]
> how about the Mac mini ? I can not run up automatically while it start up.


Disable FileVault
